packages/flutter_tools/lib/src/localizations/message_parser.dart

```diff
@@ -581,6 +585,10 @@ class Parser {
       checkExtraRules(syntaxTree);
       return syntaxTree;
     } on L10nParserException catch (error) {
+      // For debugging purposes.
+      if (logger == null) {
```

Comment:
in which case will logger be null?

Reply:
Mainly tests where I don't pass in a logger.